bpf: Quieten mock targets

joestringer opened the pull request:
Make the bpf mock testing framework targets respect the user's verbosity
flag.

joestringer committed:
bpf: Quieten mock targets

Make the bpf mock testing framework targets respect the user's verbosity
flag.

Signed-off-by: Joe Stringer <joe@cilium.io>

borkmann commented on test/bpf/Makefile:
```diff
@@ -43,16 +43,16 @@ elf-demo.o: elf-demo.c
-	$(QUIET) $(DOCKER_RUN) $(CLANG) $(FLAGS) -I../../bpf/mock -I $../../bpf/ -I../../../CMock/src -I../../../CMock/vendor/unity/src  $< ../../../CMock/vendor/unity/src/unity.c ../../../CMock/src/cmock.c ../../bpf/mock/mocks/mock_helpers.c ../../bpf/mock/mocks/mock_helpers_skb.c -o $@
+	$(QUIET)$(MAKE) -C $(ROOT_DIR)/bpf/mock generate_helper_headers
+	$(QUIET)$(MAKE) -C $(ROOT_DIR)/bpf/mock mock_helpers
+	$(QUIET)$(QUIET) $(DOCKER_RUN) $(CLANG) $(FLAGS) -I../../bpf/mock -I $../../bpf/ -I../../../CMock/src -I../../../CMock/vendor/unity/src  $< ../../../CMock/vendor/unity/src/unity.c ../../../CMock/src/cmock.c ../../bpf/mock/mocks/mock_helpers.c ../../bpf/mock/mocks/mock_helpers_skb.c -o $@
```
`$(QUIET)$(QUIET)` needed twice? Otherwis lgtm.

joestringer replied:
Woops. Fixed.

qmonnet commented on bpf/mock/Makefile:
```diff
@@ -22,13 +22,13 @@ check_helper_headers: generate_helper_headers mock_helpers
 	# Take contents from bpf/helpers.h, bpf/helpers_skb.h and bpf/helpers_xdp.h.
 	# Prune to be ready to mock and save to the current directory.
-	mkdir -p mocks; \
+	$(QUIET)mkdir -p mocks; \
```
Nit: Would be nice to indent the `sed` lines below, I didn't understand at first that their were the continuation of the same command and was wondering why they didn't get their `$(QUIET)`.

joestringer replied:
I'm actually not sure why there's all these `;` throughout this Makefile, probably the original author didn't fully understand the Makefile syntax and how this causes the rest of the commands to be shelled out and to ignore the result of each command. Would be a nice cleanup but yeah seems like we could follow up separately on this.

qmonnet replied:
Right, didn't pay attention to that but now that you mention it, they should be separate commands entirely.
Agreed that this is for a future PR :+1:.

joestringer replied:
^^ I've added a new commit on top to resolve this as well, PTAL.

qmonnet reviewed (approved):
Somehow I got triggered on the `Quieten` -- but it's not a typo this time, all good :) PR looks good as well.

joestringer committed:
bpf/Makefile: Remove unnecessary shell references

These Makefiles were sprinkled with semi-colons, causing the overall
statement to be run as a series of commands in a shell and to ignore
the result. Remove them and rely on regular Makefile statements.

Signed-off-by: Joe Stringer <joe@cilium.io>
